### ISSUE TITLE
Table toolbar is not respecting the editor.ui.viewportOffset set at runtime

### ISSUE THREAD

**Issue by niegowski** [ckeditor/ckeditor5#10597]
## 📝 Provide detailed reproduction steps (if any)

1.  https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/classic-editor.html
2.  insert a table with a lot of rows (that the table won't fit in the editor)
3.  run in the console: `editor.ui.viewportOffset = { top: 200 }`

### ✔️ Expected result

Table toolbar is not floating away out of the viewport offset as the main toolbar does

### ❌ Actual result

![](https://user-images.githubusercontent.com/1232187/134642589-07d1922e-e55f-40b9-aede-cb1d13a3a1b9.gif)

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: …
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by oleq** [comment]
Will it work properly if you blur the table and re-focus it again?

**Comment by niegowski** [comment]
> Would it work properly if you blur the table and re-focus it again?

Nope, the floating table toolbar is behaving the same as with the original `viewportOffset` and ignores the runtime change.

**Comment by oleq** [comment]
@dawidurbanski Do you know what's going on here?

**Comment by dawidurbanski** [comment]
I'll have a look

**Comment by kengreenlee** [comment]
I have the same problem.  It's also a problem with the image toolbar.  Not only does it scroll off the top, it can also scroll off the bottom.

**Comment by Reinmar** [comment]
Notes from the refinement:

*   We read this value directly from the config, while we should read if from editor.ui.viewportOffset
*   See [https://github.com/ckeditor/ckeditor5/blob/af10026601b4599442c60fad4ec5bc15b93098fd/packages/ckeditor5-ui/src/panel/balloon/contextualballoon.js#L512-L512](https://github.com/ckeditor/ckeditor5/blob/af10026601b4599442c60fad4ec5bc15b93098fd/packages/ckeditor5-ui/src/panel/balloon/contextualballoon.js#L512-L512)
*   We accept the fact that you still have to re-focus the editor to update this position. It's a different issue (has a separate ticket).
*   We should also check whether it works fine for long images.
